src/js/subtraction/components/CreateSubtraction.tsx

```diff
@@ -13,50 +13,74 @@ import {
     ViewHeader,
     ViewHeaderTitle,
 } from "../../base";
-import { useListFiles } from "../../files/querys";
+import { useInfiniteFindFiles } from "../../files/querys";
 import { FileType } from "../../files/types";
 import PersistForm from "../../forms/components/PersistForm";
-import { createSubtraction } from "../actions";
-import { SubtractionFileSelector } from "./FileSelector";
+import { create } from "../api";
+import { SubtractionFileSelector } from "./SubtractionFileSelector";
 
 const validationSchema = Yup.object().shape({
     name: Yup.string().required("A name is required"),
-    uploadId: Yup.string().required("Please select a file"),
+    uploadId: Yup.array().min(1, "Please select a file"),
 });
 
-const castValues = files => values => {
-    const uploadId = find(files, ["id", values.uploadId]) ? values.uploadId : "";
-    return { ...values, uploadId };
+type formValues = {
+    name: "";
+    nickname: "";
+    uploadId: string[];
```

Comment:
I would probably keep it as `string` and adjust the parameters you pass to the hook.